test/over_react/component_declaration/builder_integration_tests/component2/constant_required_accessor_integration_test.dart

```diff
@@ -67,62 +44,70 @@ void main() {
           attachedToDocument: true,
         );
 
-        expect(consoleErrors, isEmpty, reason: 'should not have outputted a warning but found: $consoleErrors');
+        expect(() {
+          jacket.rerender((ComponentTest()
+            ..required = true
+            ..nullable = true
+            ..requiredAndLengthLimited = [1,2]
+          )());
+        }, logsNoPropTypeWarnings);
 
-        jacket.rerender((ComponentTest()
+        expect(() {
+          jacket.rerender((ComponentTest()
             ..nullable = true
             ..requiredAndLengthLimited = [1,2]
           )()
-        );
-
-        expect(consoleErrors, isNotEmpty, reason: 'should have outputted a warning');
-        expect(consoleErrors, [contains('ComponentTestProps.required')]);
+          );
+        }, logsPropRequiredError('ComponentTestProps.required'));
       });
-
     });
 
     group('throwing when a prop is required and set to null', () {
       test('on mount', () {
-        render((ComponentTest()
+        expect(() {
+          render((ComponentTest()
             ..required = null
             ..nullable = true
             ..requiredAndLengthLimited = [1,2]
           )());
-
-        expect(consoleErrors, isNotEmpty, reason: 'should have outputted a warning');
-        expect(consoleErrors, [contains('ComponentTestProps.required')]);
+        }, logsPropRequiredError('ComponentTestProps.required'));
       });
 
       test('on re-render', () {
         var jacket = mount((ComponentTest()
-            ..required = true
-            ..nullable = true
-            ..requiredAndLengthLimited = [1,2]
-          )(),
+          ..required = true
+          ..nullable = true
+          ..requiredAndLengthLimited = [1,2]
+        )(),
           attachedToDocument: true,
         );
 
-        expect(consoleErrors, isEmpty, reason: 'should not have outputted a warning but found: $consoleErrors');
-
-        jacket.rerender((ComponentTest()
-              ..required = null
-              ..nullable = true
-              ..requiredAndLengthLimited = [1,2]
-            )());
+        expect(() {
+          jacket.rerender((ComponentTest()
+            ..required = true
+            ..nullable = true
+            ..requiredAndLengthLimited = [1,2]
+          )());
+        }, logsNoPropTypeWarnings);
```

Comment:
Why do we need to do a re-render to assert that the test is setup correctly? There is a `mount` happening with the same props above, couldn't we assert the `logsNoPropTypeWarnings` there?

Reply:
Yep, good catch - I did do that the hard way!